doc/file_services.md

```diff
@@ -0,0 +1,745 @@
+```
+
+Open multiple files
+
+``` hlasm
+         OPEN  (dcbname1,type,dcbname2,type)
+```
+
+The same, using register notation
+
+``` hlasm
+         OPEN  ((reg),type)
+         OPEN  ((reg1),type,(reg2),type)
+```
+
+#### Registers
+
+* R0 = Flags
+* R1 = DCB addresses
+
+#### Abends
+
+* S013 OPEN failed and no SYNAD exit provided
+
+### OPEN (VSE) - Open file 
+
+Open one or more files for UPDATE.
+
+#### Usage
+
+``` hlasm
+         OPEN dtfname
+         OPEN (dtfname1,dtfname2,...)
+```
+
+#### Registers
+
+* R0 = Flags
+* R1 = DTF addresses
+
+#### Abends
+
+* S013 OPEN failed and no SYNAD exit provided
+
+### CLOSE - Close file
+
+Close one or more files.
+
+#### Usage
+
+Close one file
+
+``` hlasm
+         CLOSE dcbname
+```
+
+Close multiple files
+
+``` hlasm
+         CLOSE (dcbname1,,dcbname2)
+```
+
+The same, using register notation
+
+``` hlasm
+         CLOSE ((reg1),,(reg2))
+```
+
+#### Registers
+
+* R1 = DCB addresses
+
+#### Abends
+
+* S013 CLOSE failed and no SYNAD exit provided
+
+### CLOSE (VSE) - Close file
+
+Close one or more files.
+
+#### Usage
+
+``` hlasm
+         CLOSE (dtfname1,dtfname2,...)
+```
+
+#### Registers
+
+* R1 = DTF addresses
+
+#### Abends
+
+* S013 CLOSE failed and no SYNAD exit provided
+
+### GET - Read sequential record
+
+Read a sequential record
+
+#### Usage
+
+Read a record into an I/O area supplied on DCB RECORD=ioarea
+    
+``` hlasm
+         GET dcbname
+```
+
+Read a record into an I/O area
+
+``` hlasm
+         GET dcbname,ioarea
+```
+
+The same, using register notation
+
+``` hlasm
+         GET (reg)
+         GET (reg1),(reg2)
+```
+#### Registers
+
+* R0 = ioarea
+* R1 = DCB address
+
+### PUT - Write a sequential record
+
+Write a sequential record
+
+#### Usage
+
+Write a record from an I/O area supplied on the DCB RECORD=ioarea
+
+``` hlasm
+         PUT dcbname
+```
+
+Write a record from an I/O area
+
+``` hlasm
+         PUT dcbname,ioarea
+```
+
+The same, using register notation
+
+``` hlasm
+         PUT (reg)
+         PUT (reg1),(reg2)
+```
+
+#### Registers
+
+* R0 = ioarea
+* R1 = DCB address
+
+### READ - Read block from file
+
+ Read a block from a file.
+
+ 
+1. If EODAD or SYNAD result from the READ, they will only be
+   processed by the CHECK macro.
+2. If the decbname parameter is specified in register notation,
+   no internal DECB is generated. It is the programmer's
+   responsibility to create and address the 16-byte DECB.
+ 
+#### Usage
+ 
+Read a block using an internal DECB, into an I/O area
+supplied on the DCB RECORD=ioarea
+
+``` hlasm
+         READ decbname,,dcbname
+```
+
+Read a block using an internal DECB into an I/O area.
+The I/O area must be at least as large as DCB BLKSIZE
+
+``` hlasm
+         READ decbname,,dcbname,ioarea
+```
+
+The same, using register notation
+
+``` hlasm
+         READ (reg1),,(reg2)
+         READ (reg1),,(reg2),(reg3)
+```
+
+#### Registers
+
+* R0 = DCB and ioarea addresses...saved
```

Comment:
@abekornelis @dsh33782 Any ideas on this.. I just took this as is from the existing doco.

Reply:
@abekornelis @dsh33782 Can you let me know what the following means in terms of the documentation:

> R0 = DCB and ioarea addresses...saved

@jyganci is not sure what the ...saved means at the end.

Reply:
I'm not sure at all what the intended meaning may have been. I've also lost context. Which function is being described here? I tried to follow the link on the line numbers, but it doesn't do anything (in my browser (chrome)). Would you please provide a little more context? I then could have a look at the actual java code to validate what this piece of text is trying to get across.

Reply:
Larger context from the documentation.

### WRITE - Write block to file

Write a block to a file.

1. If SYNAD results from the WRITE, it will only be processed
   by the CHECK macro.
2. If the decbname parameter is specified in register notation,
   no internal DECB is generated. It is the programmer's
   responsibility to create and address the 16-byte DECB.

#### Usage

Write a block using an internal DECB, from an I/O area supplied on the DCB RECORD=ioarea

``` hlasm
         WRITE decbname,,dcbname
```

Write a block using an internal DECB from an I/O area

``` hlasm
         WRITE decbname,,dcbname,ioarea
```

The same, using register notation

``` hlasm
         WRITE (reg1),,(reg2)
         WRITE (reg1),,(reg2),(reg3)
```

#### Registers

* **R0 = DCB and ioarea addresses...saved**
* R1 = DECB address
* R15= DCB address

The line describing R0 above is the subject of this issue..

**R0 = DCB and ioarea addresses...saved**

The question is, what does the ...saved bit mean?

Reply:
Here's what should be on that line:

    R0 = work register

The remainder of this comment explains what the original line meant -- no need to put any of it in the doc.

Internal to the WRITE macro, which invokes the ZWRITE macro, R0 is used to store the addresses of the DCB and the I/O area in the DECB. That's what the "...saved" means. Definitely does not belong in the doc.

Reply:
Doc has been updated as suggested by @jyganci. See bf48643